packages/flutter/lib/src/material/dialog.dart

```diff
@@ -918,19 +935,21 @@ Future<T> showDialog<T>({
     context: context,
     pageBuilder: (BuildContext buildContext, Animation<double> animation, Animation<double> secondaryAnimation) {
       final Widget pageChild = child ?? Builder(builder: builder);
-      return SafeArea(
-        child: Builder(
-          builder: (BuildContext context) {
-            return theme != null
-                ? Theme(data: theme, child: pageChild)
-                : pageChild;
-          }
-        ),
+      Widget dialog = Builder(
+        builder: (BuildContext context) {
+          return theme != null
+            ? Theme(data: theme, child: pageChild)
+            : pageChild;
+        }
       );
+      if (useSafeArea) {
+        dialog = SafeArea(child: dialog);
+      }
+      return dialog;
     },
     barrierDismissible: barrierDismissible,
     barrierLabel: MaterialLocalizations.of(context).modalBarrierDismissLabel,
-    barrierColor: Colors.black54,
+    barrierColor: barrierColor ?? Colors.black54,
```

Comment:
What do you think about adding `barrierColor` to `DialogTheme` as well?

Reply:
Yup. It's on my list. I want to update DialogTheme to bring it in line with the other themes first and then add a few more parameters like this (and the insetsPadding and clipBehavior we added a while back).